### PR TITLE
endpoint api/users/current with preloader

### PR DESCRIPTION
**Pull request by javasia**
backend part and test

### CODE REVIEW

**Comment by javasia** [comment]
[2h - Endpoint "api/users/current" which will always return the same one user (a-la currently logged in). Use it to load user data on front-end upon application load. Store loaded user in Redux](https://trello.com/c/T6U1qwW1/76-2h-endpoint-api-users-current-which-will-always-return-the-same-one-user-a-la-currently-logged-in-use-it-to-load-user-data-on-fr)

File: src/main/java/com/danit/finalproject/application/controller/UserController.java [hunk 1]
```diff
@@ -32,6 +34,12 @@ public User getUserById(@PathVariable Long userId) {
     return userService.getUserById(userId);
   }
 
+  @GetMapping("current")
+  @ResponseBody
```
Comment (skosinskiy) [inline src/main/java/com/danit/finalproject/application/controller/UserController.java]: ResponseBody annotation in not neccessary for @RestController. The difference between Controller and RestController is that RestController contains ResponseBody annotation

File: src/main/java/com/danit/finalproject/application/controller/UserController.java [hunk 2]
```diff
@@ -32,6 +34,12 @@ public User getUserById(@PathVariable Long userId) {
...
+  public User getCurrentUser(@CookieValue(value = "user_id") String userCookie) {
```
Comment (skosinskiy) [inline src/main/java/com/danit/finalproject/application/controller/UserController.java]: You don't have to pass cookie value into controller as param. Remove it from parameters. Return always the same User (select one you want from data.sql). In future when auth mechanism will be implemented logged user will be passed as argument into this Controller by Spring Container, you don't have to manually pass cookie to find logged user.

File: src/test/java/com/danit/finalproject/application/controller/UserControllerTest.java [hunk 3]
```diff
@@ -55,6 +62,37 @@ public void getUserById() throws Exception {
 
 	}
 
+  @Test
+  public void getCurrentUserWithCookie() throws Exception {
```
Comment (skosinskiy) [inline src/test/java/com/danit/finalproject/application/controller/UserControllerTest.java]: tests should be simplified according to comments above

**Comment by javasia** [comment]
[2SP - Endpoint "api/users/current" which will always return the same one user (a-la currently logged in). Use it to load user data on front-end upon application load. Store loaded user in Redux. Show preloader upon initial user fetch from server](https://trello.com/c/T6U1qwW1/76-2sp-endpoint-api-users-current-which-will-always-return-the-same-one-user-a-la-currently-logged-in-use-it-to-load-user-data-on-f)

**Pull request merged**